In [4]:
import pandas as pd

In [5]:
from statsmodels.sandbox.regression.gmm import IV2SLS

In [6]:
# Load the dataset
file_path = '/Users/shivamverma/Downloads/midterm_partone.csv'

In [7]:
data = pd.read_csv(file_path)


In [8]:
# Define the dependent variable (Y) and independent variables (X)
# Dependent variable: Stock Change
Y = data['Stock Change']


In [9]:
# Independent variables: assuming Inventory Turnover, Operating Profit, etc., as X
X = data[['Constant', 'Inventory Turnover', 'Operating Profit', 'Interaction Effect',
          'Current Ratio', 'Quick Ratio', 'Debt Asset Ratio']]


In [10]:
# Instrumental variables (IVs), assuming the same as independent variables
Z = X


In [11]:
# Create the model with the specified endogenous and instrumental variables
# Incorporate the moment condition modification to test for bias
model = IV2SLS(Y, X, instrument=Z)
result = model.fit()

In [12]:
# Display the summary of the GMM estimation
print(result.summary())


                          IV2SLS Regression Results                           
Dep. Variable:           Stock Change   R-squared:                       0.015
Model:                         IV2SLS   Adj. R-squared:                  0.012
Method:                     Two Stage   F-statistic:                     4.364
                        Least Squares   Prob (F-statistic):           0.000221
Date:                Mon, 11 Nov 2024                                         
Time:                        07:11:33                                         
No. Observations:                1696                                         
Df Residuals:                    1689                                         
Df Model:                           6                                         
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
Constant               0.0401      0

# Based on the IV2SLS regression results, here’s a summary of key findings:

* Significant Coefficients:

Operating Profit and Interaction Effect are statistically significant at the 1% level, with 
𝑝
p-values of 0.000.
Operating Profit has a negative coefficient, indicating that an increase in operating profit is associated with a decrease in stock change, holding other factors constant.
Interaction Effect has a positive coefficient, suggesting a direct positive relationship with stock change.

* Insignificant Coefficients:

Constant, Inventory Turnover, Current Ratio, Quick Ratio, and Debt Asset Ratio are not statistically significant. Their 𝑝-values are above the typical 5% threshold, implying weak evidence that these variables independently affect stock change.

* Model Fit:

The R² value of 0.015 suggests that the model explains only 1.5% of the variance in stock changes, indicating a weak overall fit.

* Implications for 𝛿 Term (Bias in Moment Conditions):

The lack of significance for most coefficients suggests limited evidence of the claimed bias (𝛿) across all instrumental variables. However, the significance of Operating Profit and Interaction Effect might suggest specific relationships, though these may not necessarily indicate a pervasive bias as claimed by the industry expert.

In summary, the data does not strongly support the expert's claim of a systematic bias across all moment conditions in the instrumental variables.






